toolchain/lower/mangler.h

```diff
@@ -0,0 +1,48 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_TOOLCHAIN_LOWER_MANGLER_H_
+#define CARBON_TOOLCHAIN_LOWER_MANGLER_H_
+
+#include <string>
+
+#include "toolchain/lower/file_context.h"
+#include "toolchain/sem_ir/ids.h"
+
+namespace Carbon::Lower {
+
+// A class for producing mangled (deterministically unique, at least partially
+// human readable) names for externally referenceable entities such as extern
+// functions.
```

Comment:
```suggestion
// human readable) names for externally referenceable entities such as
// functions.
```

"extern" has what I think is a slightly different meaning in Carbon than in C++, would it make sense to omit it here?